ports/raspberrypi/supervisor/port.c

```diff
@@ -257,15 +256,31 @@ void port_heap_init(void) {
 }
 
 void *port_malloc(size_t size, bool dma_capable) {
+    if (!dma_capable && _psram_size > 0) {
+        void *block = tlsf_malloc(_psram_heap, size);
+        if (block) {
+            return block;
+        }
+    }
     void *block = tlsf_malloc(_heap, size);
     return block;
 }
 
 void port_free(void *ptr) {
-    tlsf_free(_heap, ptr);
+    if (((size_t)ptr) < SRAM_BASE) {
```

Comment:
This is kind of tricky, because it involves knowing where the PSRAM base and the SRAM base are. But there's plenty of other magic here, like `0x11000000`. Unfortunately there don't seem to be good `#define`s for this in the pico-sdk header files. There is `XIP_NOALLOC_BASE` for RP2040, but that's _not_ the right value for RP2350 start of PSRAM.